src/main/java/jcifs/smb1/smb1/Dfs.java

```diff
@@ -0,0 +1,328 @@
+/* jcifs smb client library in Java
+ * Copyright (C) 2008  "Michael B. Allen" <jcifs at samba dot org>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs.smb1.smb1;
+
+import java.util.*;
+
+import jcifs.smb1.Config;
+import jcifs.smb1.UniAddress;
+import jcifs.smb1.util.*;
+
+import java.io.*;
+
+public class Dfs {
+
+    static class CacheEntry {
+        long expiration;
+        HashMap map;
+
+        CacheEntry(long ttl) {
+            if (ttl == 0)
+                ttl = Dfs.TTL;
+            expiration = System.currentTimeMillis() + ttl * 1000L;
+            map = new HashMap();
+        }
+    }
+
+    static LogStream log = LogStream.getInstance();
+    static final boolean strictView = Config.getBoolean("jcifs.smb1.smb.client.dfs.strictView", false);
+    static final long TTL = Config.getLong("jcifs.smb1.smb.client.dfs.ttl", 300);
+    static final boolean DISABLED = Config.getBoolean("jcifs.smb1.smb.client.dfs.disabled", false);
+
+    protected static CacheEntry FALSE_ENTRY = new Dfs.CacheEntry(0L);
+
+    protected CacheEntry _domains = null; /* aka trusted domains cache */
+    protected CacheEntry referrals = null;
+
+    public HashMap getTrustedDomains(NtlmPasswordAuthentication auth) throws SmbAuthException {
+        if (DISABLED || auth.domain == "?")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.